nexus/src/sagas.rs

```diff
@@ -1109,12 +1109,53 @@ async fn sdc_create_disk_record(
     // but this should be acceptable because the disk remains in a "Creating"
     // state until the saga has completed.
     let volume_id = sagactx.lookup::<Uuid>("volume_id")?;
+    let opctx = OpContext::for_saga_action(&sagactx, &params.serialized_authn);
+
+    let block_size: db::model::BlockSize = match &params
+        .create_params
+        .disk_source
+    {
+        params::DiskSource::Blank { block_size } => {
+            db::model::BlockSize::try_from(*block_size).map_err(|e| {
+                ActionError::action_failed(Error::internal_error(
+                    &e.to_string(),
```

Comment:
maybe want a little more context in this message?  I'm not sure what should go here -- I'm just imagining we see this error in the log.  We'll be able to tell it came from this saga action, at least.

Reply:
we'll at least see `invalid block size {}` from the `db::model::BlockSize` `try_from`, though I'm not sure how we would ever hit this error because `params::BlockSize` `try_from` also validates that it's one of the (currently) three supported values?